API/controllers/article-controller.js

```diff
@@ -27,3 +31,27 @@ exports.getArticles = (req, res, next) => {
     res.status(200).send({ articles });
   });
 };
+
+exports.patchArticle = (req, res, next) => {
+  const { inc_votes } = req.body;
+  const { article_id } = req.params;
+
+  if (isNaN(Number(inc_votes))) next({ status: 400, msg: "Malformed Body" });
+
+  return fetchArticle(article_id)
+    .then(({ rows }) => {
+      return new Promise((resolve, reject) => {
+        if (rows.length === 0)
+          reject({ status: 404, msg: "Article Not Found" });
+        else resolve(rows[0]);
+      });
+    })
```

Comment:
What is the purpose of this `fetchArticle` query ?
If you execute the query to UPDATE article, and include a `RETURNING *`, you can check if rows are empty there and spare an unnecessary query 🤔 

I'm also unsure of this approach of throwing an error from the controller, rather than the model. This is a debatable topic, to me though, it seems more efficient to throw the error for empty `rowCount` from the _model_, because it's the model's job to handle and validate database queries.

If it throws in the model, it will go directly to the _controller's_ `catch`. Whereas this way, it is going to the _controller's_ `then` first, which is an extra step in the journey